src/t8_data/t8_shmem.h

```diff
@@ -147,6 +147,32 @@ void                t8_shmem_array_allgather (const void *sendbuf,
                                               int recvcount,
                                               sc_MPI_Datatype recvtype);
 
+/**
+ * Fill a t8_shmem array with an Allgatherv
+ * Computes the recvcount-array and displacement-array for each rank of a node using the
+ * sendcount.
+ * The total number of items of each node is then used to compute the
+ * recvcount-array and displacement-array between nodes. 
```

Comment:
Since `MPI_Allgatherv` does not scale well in HPC (and t8code is especially written for the HPC context), we should add a disclaimer to use `MPI_Allgather` whenever possible.

Reply:
I don't agree with that. The name of the function is similar to the MPI-name and the user should be aware of the scaling behaviour. You're right, All-to-all communication does not scale well, but sometime we can't avoid it. Nevertheless we don't warn the user about it.

Reply:
The negative consequence would be to comment on all functions of t8code how they scale. We don't warn the user at the function `t8_shmem_array_allgather' about the scaling and I don't think we should here either.